airflow/providers/amazon/aws/hooks/sns.py

```diff
@@ -24,6 +24,20 @@
 from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
 
 
+def _get_message_attribute(o):
+    if isinstance(o, bytes):
+        return {'DataType': 'Binary', 'BinaryValue': o}
+    elif isinstance(o, str):
+        return {'DataType': 'String', 'StringValue': o}
+    elif isinstance(o, (int, float)):
+        return {'DataType': 'Number', 'StringValue': str(o)}
+    elif hasattr(o, '__iter__'):
+        return {'DataType': 'String.Array', 'StringValue': json.dumps(o)}
+    else:
+        raise TypeError('Values in MessageAttributes must be one of bytes, str, int, float, or iterable; '
+                        f'got {type(o)}')
```

Comment:
You can remove the `else:`

and just keep:
`raise TypeError('Values in MessageAttributes must be one of bytes, str, int, float, or iterable; '
                        f'got {type(o)}')`